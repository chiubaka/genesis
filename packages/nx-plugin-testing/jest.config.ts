export default {
  displayName: "nx-plugin-testing",
  preset: "../../jest.preset.js",
  globals: {
    "ts-jest": {
      tsconfig: "<rootDir>/tsconfig.spec.json",
    },
  },
  transform: {
    "^.+\\.[tj]s$": "ts-jest",
  },
  moduleFileExtensions: ["ts", "js", "html"],
  coverageDirectory: "../../reports/coverage/packages/nx-plugin-testing",
  reporters: [
    "default",
    [
      "jest-junit",
      {
        addFileAttribute: "true",
        classNameTemplate: "{suitename}",
        outputDirectory: "reports/junit",
        outputName: "nx-plugin-testing.xml",
      },
    ],
  ],
};
