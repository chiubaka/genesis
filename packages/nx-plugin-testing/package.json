{
  "name": "@chiubaka/nx-plugin-testing",
  "version": "0.3.0",
  "description": "Utilities for testing Nx plugins",
  "main": "src/index.js",
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/chiubaka/nx-plugin.git"
  },
  "keywords": [
    "nx",
    "nx-plugin",
    "testing",
    "e2e"
  ],
  "author": "Daniel Chiu <daniel@chiubaka.com>",
  "license": "UNLICENSED",
  "bugs": {
    "url": "https://github.com/chiubaka/nx-plugin/issues"
  },
  "homepage": "https://github.com/chiubaka/nx-plugin#readme",
  "dependencies": {
    "@nx/devkit": "16.8.1",
    "@nx/node": "16.8.1",
    "nx": "16.8.1"
  },
  "types": "./src/index.d.ts"
}
