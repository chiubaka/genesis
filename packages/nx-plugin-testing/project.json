{
  "name": "nx-plugin-testing",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/nx-plugin-testing/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/nx-plugin-testing",
        "main": "packages/nx-plugin-testing/bin/index.ts",
        "tsConfig": "packages/nx-plugin-testing/tsconfig.lib.json",
        "assets": [
          "packages/nx-plugin-testing/*.md",
          {
            "input": "./packages/nx-plugin-testing/bin",
            "glob": "**/!(*.ts)",
            "output": "./bin"
          },
          {
            "input": "./packages/nx-plugin-testing/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          },
          {
            "input": "./packages/nx-plugin-testing",
            "glob": "generators.json",
            "output": "."
          },
          {
            "input": "./packages/nx-plugin-testing",
            "glob": "executors.json",
            "output": "."
          }
        ],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "deploy": {
      "executor": "@chiubaka/nx-plugin:npm-publish",
      "defaultConfiguration": "development",
      "dependsOn": [
        {
          "target": "build"
        },
        {
          "target": "deploy:remove-yarn-module"
        }
      ],
      "options": {
        "packagePath": "dist/packages/nx-plugin-testing"
      },
      "configurations": {
        "production": {},
        "development": {
          "registryUrl": "http://localhost:4873",
          "skipLogin": false,
          "skipUnpublish": false
        }
      }
    },
    "deploy:remove-yarn-module": {
      "executor": "nx:run-commands",
      "options": {
        "command": "rm -rf node_modules/@chiubaka/nx-plugin"
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/nx-plugin-testing"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": [
        "{workspaceRoot}/reports/coverage/packages/nx-plugin-testing",
        "{workspaceRoot}/reports/junit/nx-plugin-testing.xml"
      ],
      "options": {
        "jestConfig": "packages/nx-plugin-testing/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
