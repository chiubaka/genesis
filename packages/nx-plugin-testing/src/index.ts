export * from "./setup";
export * from "./testingWorkspace";
export * from "./verdaccio";
