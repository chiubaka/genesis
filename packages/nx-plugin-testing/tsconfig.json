{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "types": ["jest", "node"]
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
