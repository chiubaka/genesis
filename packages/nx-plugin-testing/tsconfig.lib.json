{
  "extends": "./tsconfig.json",
  "include": ["**/*.ts"],
  "exclude": ["jest.config.ts", "**/*.spec.ts", "**/*.test.ts", "test"]
}
