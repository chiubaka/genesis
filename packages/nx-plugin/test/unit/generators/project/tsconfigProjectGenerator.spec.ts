import { createTreeWithLibWorkspace } from "@chiubaka/nx-plugin-testing";

import {
  Project,
  TsConfigGeneratorPresets,
  tsconfigProjectGenerator,
} from "../../../../src";
import { tsconfigTestCases } from "../../../cases";

describe("tsconfigProjectGenerator", () => {
  const projectName = "tsconfig-project";
  const projectType = "library";

  let project: Project;

  const getProject = () => {
    return project;
  };

  describe("without any options", () => {
    beforeAll(async () => {
      const tree = await createTreeWithLibWorkspace(projectName);
      tsconfigProjectGenerator(tree, {
        projectName,
        projectType,
      });
      project = new Project(tree, projectName, projectType);
    });

    tsconfigTestCases(getProject);
  });

  describe("with Node 18 preset", () => {
    beforeAll(async () => {
      const tree = await createTreeWithLibWorkspace(projectName);
      tsconfigProjectGenerator(tree, {
        projectName,
        projectType,
        ...TsConfigGeneratorPresets.node18,
      });
      project = new Project(tree, projectName, projectType);
    });

    tsconfigTestCases(getProject, {
      appLibTypes: ["node"],
      compilerOptions: {
        lib: ["es2022"],
        module: "commonjs",
        target: "es2022",
      },
    });
  });
});
