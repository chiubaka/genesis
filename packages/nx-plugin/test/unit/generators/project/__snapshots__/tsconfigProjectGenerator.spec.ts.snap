// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsconfigProjectGenerator with Node 18 preset tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "declaration": true,
    "types": Array [
      "node",
    ],
  },
  "exclude": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": Array [
    "**/*.ts",
  ],
}
`;

exports[`tsconfigProjectGenerator with Node 18 preset tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "lib": Array [
      "es2022",
    ],
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "target": "es2022",
  },
  "extends": "../../tsconfig.base.json",
  "files": Array [],
  "include": Array [],
  "references": Array [
    Object {
      "path": "./tsconfig.lib.json",
    },
    Object {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`tsconfigProjectGenerator with Node 18 preset tsconfig.spec.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "types": Array [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.d.ts",
    "test",
  ],
}
`;

exports[`tsconfigProjectGenerator without any options tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "declaration": true,
  },
  "exclude": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": Array [
    "**/*.ts",
  ],
}
`;

exports[`tsconfigProjectGenerator without any options tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "outDir": "../../dist/out-tsc",
  },
  "extends": "../../tsconfig.base.json",
  "files": Array [],
  "include": Array [],
  "references": Array [
    Object {
      "path": "./tsconfig.lib.json",
    },
    Object {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`tsconfigProjectGenerator without any options tsconfig.spec.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "types": Array [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.d.ts",
    "test",
  ],
}
`;
